contracts/prover/HyperProver.sol

```diff
@@ -82,12 +104,73 @@ contract HyperProver is IMessageRecipient, BaseProver, Semver {
             revert UnauthorizedDispatch(sender);
         }
 
+        (MessageType messageType, bytes memory messageBody) = _decodeMessage(
+            _message
+        );
+
+        if (messageType == MessageType.STANDARD) {
+            return handleStandardMessage(messageBody);
+        } else if (messageType == MessageType.COMPRESSED) {
+            return handleCompressedMessage(messageBody);
+        } else {
+            revert InvalidMessageType(messageType);
+        }
+    }
+
+    /**
+     * @dev Handles the details of how to handle standard messages.
+     * @param message The bytes memory type message.
+     */
+    function handleStandardMessage(bytes memory message) internal {
         // Decode message containing intent hashes and claimants
         (bytes32[] memory hashes, address[] memory claimants) = abi.decode(
-            _messageBody,
+            message,
             (bytes32[], address[])
         );
 
+        proveBatch(hashes, claimants);
+    }
+
+    /**
+     * @dev Handles the details of how to handle compressed messages.
+     * @param _messageBody The body of the message in bytes memory type.
+     */
+    function handleCompressedMessage(bytes memory _messageBody) internal {
```

Comment:
@carlosfebres isn't it the same amount of gas to call this function as just send through Hyperlane?  The gas you are saving on the source is this calldata cost. Does hyperlane charges % on top?

Reply:
The goal isn't to reduce the total gas consumption but to reduce the gas cost of relaying messages through Hyperlane. AFAIK they don't change anything on top (apart from the absurd gas prices they use).

Reply:
I see, that makes sense. Their gas prices make it extra expensive to relay through them